### PR TITLE
Issue #19112: Remove old tabs tray and synced tabs code

### PR DESCRIPTION
**Pull request by jonalmeida**
🔥

### CODE REVIEW

**Comment by jonalmeida** [comment]
Let's wait until the beta cut to land this. 🙂

**Comment by jonalmeida** [comment]
How have these tests been passing when they pointed to the wrong ID.. 🤔 
```diff
diff --git a/app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt b/app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt
index 32fc8d255..b9e36968b 100644
--- a/app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt
@@ -666,7 +666,7 @@ class DefaultSessionControlControllerTest {

         verify {
             navController.navigate(
-                match<NavDirections> { it.actionId == R.id.action_global_tabTrayDialogFragment },
+                match<NavDirections> { it.actionId == R.id.action_global_tabsTrayFragment },
                 null
             )
         }
diff --git a/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt b/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt
index 7f61b5b0f..2d000abd6 100644
--- a/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt
@@ -124,7 +124,7 @@ class DefaultTabsTrayControllerTest {
         verifyOrder {
             profiler.getProfilerTime()
             navController.navigateBlockingForAsyncNavGraph(
-                TabTrayDialogFragmentDirections.actionGlobalHome(focusOnAddressBar = true)
+                TabsTrayFragmentDirections.actionGlobalHome(focusOnAddressBar = true)
             )
             navigationInteractor.onTabTrayDismissed()
             profiler.addMarker(
@@ -159,7 +159,7 @@ class DefaultTabsTrayControllerTest {
         verifyOrder {
             profiler.getProfilerTime()
             navController.navigateBlockingForAsyncNavGraph(
-                TabTrayDialogFragmentDirections.actionGlobalHome(focusOnAddressBar = true)
+                TabsTrayFragmentDirections.actionGlobalHome(focusOnAddressBar = true)
             )
             navigationInteractor.onTabTrayDismissed()
             profiler.addMarker(
```

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt
Comment (eliserichards) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:880]:
```suggestion
    @Ignore("Enable after https://github.com/mozilla-mobile/fenix/issues/19738 and https://github.com/mozilla-mobile/fenix/issues/19090 land.")
```

**Review by eliserichards — COMMENTED** [review]
I love seeing all of this red! 🙌

**Review by gabrielluong — COMMENTED** [review]
I think this looks good and probably is ready for approval. From playing around with the new tabs tray and my experience with the old tabs tray, it's easy for me to point out that there are clearly things that are missing in the new tabs tray. I think I would prefer to hold this back a little longer until we do capture all the remaining issues either with the help of engineers + Product/UX. It seems easiest to do this if we can continue to toggle on/off the new/old tabs trays for comparison and reviewing the old code.

File: app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt
Comment (jonalmeida) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SmokeTest.kt:880]: I followed the format that already existed in the other tests. :)

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayController.kt:114]: This is already moved over to the [new tabs tray](https://searchfox.org/mozilla-mobile/rev/2bc7d8d4f6aaedae4bf21e35e0bb51a37a8d5905/fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt#83), I don't think we need to do anything else.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt:288]: These are still needed for the new tabs tray. See https://github.com/mozilla-mobile/fenix/issues/19090

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:81]: AS was crapping out when I tried to do that in this PR. I guess my build directory still has references to the old naming. Happy to do it in a follow-up.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:523]: Nice find! I filed https://github.com/mozilla-mobile/fenix/issues/19784 for this.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:653]: Same. Filed https://github.com/mozilla-mobile/fenix/issues/19784.

File: app/src/main/java/org/mozilla/fenix/tabtray/FenixTabsAdapter.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/FenixTabsAdapter.kt:103]: We don't have to do most of the things that the old code was doing because we get that support for free now.

File: app/src/main/java/org/mozilla/fenix/tabtray/SaveToCollectionsButtonAdapter.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/SaveToCollectionsButtonAdapter.kt:24]: This code was used for things that are not related to "save to collections". We also do not want the 'Select tabs' feature any more https://github.com/mozilla-mobile/fenix/issues/18656.

**Comment by jonalmeida** [comment]
> From playing around with the new tabs tray and my experience with the old tabs tray, it's easy for me to point out that there are clearly things that are missing in the new tabs tray.

Which ones are you referring to? Let's file new bugs for regressions. I've only found one so far that has not been filed already.

> I think I would prefer to hold this back a little longer until we do capture all the remaining issues either with the help of engineers + Product/UX. It seems easiest to do this if we can continue to toggle on/off the new/old tabs trays for comparison and reviewing the old code.

There aren't any glaring reasons to keep around the old code, and in the off changes that we need to need to compare it to old, we can always go back in git history to build/compare. The sooner remove the old code earlier in nightly the quicker we can find more bugs that we didn't notice when the toggle was still available.

I would also like to land this patch sooner than later before it goes stale. 😄

**Comment by jonalmeida** [comment]
`run-ui` action crashed.

**Comment by gabrielluong** [comment]
For the new issues that you filed, can you also reference their respective strings that still remains? Perhaps we should point to searchfox around where I pointed out the unused strings and the surrounding functionality to provide the most context in these new issues. My concern here is that we don't have enough details in the new issues and we don't specify the strings that can be reused and therefore we run into scenarios where we are asking or adding new strings b/c we are unaware of the existing strings and how they are used.

I also saw some comments resolved without any response. I am wondering if they were just already captured in your spreadsheet or the strings/files could've been removed, but it would be better if there was just a response to everything since I left like 13 comments and I don't have to wonder if they were just ignored.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:689]: I missed this one. Filed https://github.com/mozilla-mobile/fenix/issues/19792.

File: app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabtray/TabTrayView.kt:693]: Same as above: https://github.com/mozilla-mobile/fenix/issues/19792

**Comment by jonalmeida** [comment]
> My concern here is that we don't have enough details in the new issues and we don't specify the strings that can be reused and therefore we run into scenarios where we are asking or adding new strings b/c we are unaware of the existing strings and how they are used.

I've assign them to myself as follow-up. 🙂 

> I also saw some comments resolved without any response. I am wondering if they were just already captured in your spreadsheet or the strings/files could've been removed, but it would be better if there was just a response to everything since I left like 13 comments and I don't have to wonder if they were just ignored.

Missed two of them. 👌

**Review by gabrielluong — APPROVED** [review]
I have reference some of my comments in the newly filed issues to ensure we do provide some code context and ensure we reuse the strings to avoid keeping unused strings around in our codebase.

**Pull request merged**
